### PR TITLE
Snippets for dart-mode

### PR DESCRIPTION
**Pull request by l7ssha**
Hello,
I've added few snippets for dart-mode. For now there no dart-mode snippets.

### CODE REVIEW

File: snippets/dart-mode/for [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# -*- mode: snippet -*-
+# name: for
+# key: for
+# expand-env: ((yas-indent-line 'fixed) (yas-wrap-around-region nil))
+# --
+for(var i = ${1:0}; i ${2:< 10}; i${3:++}) {
+  ${4:Body}
```
Comment (AndreaCrotti) [inline snippets/dart-mode/for]: could we not put $0 here instead of at the end of the for?

File: snippets/dart-mode/fori [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# -*- mode: snippet -*-
+# name: fori
+# key: fori
+# expand-env: ((yas-indent-line 'fixed) (yas-wrap-around-region nil))
+# --
+for(var ${1:obj} in ${2:collection}) {
```
Comment (AndreaCrotti) [inline snippets/dart-mode/fori]: same thing here

File: snippets/dart-mode/func [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: func
+# key: func
```
Comment (AndreaCrotti) [inline snippets/dart-mode/func]: maybe `fun` or even just `f` is better?

File: snippets/dart-mode/funca [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: funca
+# key: funca
```
Comment (AndreaCrotti) [inline snippets/dart-mode/funca]: considering it's an async function, maybe `afun` / `af` would seem more mnemonic to me, what do you think?

File: snippets/dart-mode/getset [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: getset
+# key: getset
+# --
+${1:Type} _${2:Name};
```
Comment (AndreaCrotti) [inline snippets/dart-mode/getset]: is that such a common pattern to have it this way?

File: snippets/dart-mode/getter [hunk 6]
```diff
@@ -0,0 +1,6 @@
+# -*- mode: snippet -*-
+# name: getter
+# key: get
+# --
```
Comment (AndreaCrotti) [inline snippets/dart-mode/getter]: same question here

**Comment by AndreaCrotti** [comment]
Great thanks @l7ssha left a few comments and questions

**Comment by l7ssha** [comment]
I've applied some of your notes @AndreaCrotti . Thanks for review.

File: snippets/dart-mode/getset (hunk 5 above)
Comment (l7ssha) [reply snippets/dart-mode/getset]: Yes, getters are commonly used.  This is generic pattern but it applies pretty well in this case.

File: snippets/dart-mode/getter (hunk 6 above)
Comment (l7ssha) [reply snippets/dart-mode/getter]: Same here

File: snippets/dart-mode/getset (hunk 5 above)
Comment (l7ssha) [reply snippets/dart-mode/getset]: And if you mean '_Name' notation - '_' in Dart means 'private' in C like programming languages, so it has to be like that.

**Comment by AndreaCrotti** [comment]
Great thanks

**Pull request merged**